packages/core-app-api/src/apis/implementations/AppTranslationApi/AppTranslationImpl.ts

```diff
@@ -136,18 +139,13 @@ export class AppTranslationApiImpl implements AppTranslationApi {
     const namespace = translationRef.getId();
     const lazyResources = initResources || translationRef.getLazyResources();
 
-    const fallbackLanguages = services.languageUtils.getFallbackCodes(
-      options.fallbackLng,
-      currentLanguage,
-    ) as string[];
-
-    Promise.allSettled(
-      [...fallbackLanguages, currentLanguage].map(addLanguage),
-    ).then(results => {
-      if (results.some(result => result.status === 'fulfilled')) {
-        this.i18n.emit('loaded');
-      }
-    });
+    Promise.allSettled((options.supportedLngs || []).map(addLanguage)).then(
```

Comment:
supportLanguages maybe `['zh', 'en']` and `fallbackLanguages` just to be 'en'. When I add `zh` resources in 'createApp', need to add zh resource to i18next context